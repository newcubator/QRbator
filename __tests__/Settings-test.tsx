import { render } from "@testing-library/react-native";
import { Alert } from "react-native";
import SettingsScreen from "../app/settings";

jest.mock("expo-sharing", () => ({
  isAvailableAsync: jest.fn().mockResolvedValue(true),
  shareAsync: jest.fn().mockResolvedValue(undefined),
}));

jest.mock("expo-file-system", () => ({
  cacheDirectory: "file://test-cache-dir/",
  writeAsStringAsync: jest.fn().mockResolvedValue(undefined),
  readAsStringAsync: jest.fn().mockResolvedValue("[]"),
}));

jest.mock("expo-document-picker", () => ({
  getDocumentAsync: jest.fn().mockResolvedValue({
    canceled: true,
    assets: [],
  }),
}));

jest.mock("~/core/qrCodeStorage", () => ({
  getAllQRCodes: jest.fn().mockResolvedValue([]),
  importQRCodes: jest.fn(),
  deleteAllQRCodes: jest.fn(),
}));

jest.spyOn(Alert, "alert");

describe("<SettingsScreen />", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with export options", () => {
    const { getByText } = render(<SettingsScreen />);

    expect(getByText("exportData")).toBeTruthy();
    expect(getByText("exportAllJson")).toBeTruthy();
    expect(getByText("exportAllCsv")).toBeTruthy();
  });

  test("matches snapshot", () => {
    const tree = render(<SettingsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
