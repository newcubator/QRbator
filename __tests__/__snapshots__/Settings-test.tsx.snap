// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> matches snapshot 1`] = `
<View
  className="flex-1 bg-corp-white px-6 py-4"
>
  <View
    className="mb-8"
    entering={{}}
  >
    <Text
      className="mb-4 text-lg font-medium text-corp-grey"
    >
      exportData
    </Text>
    <View
      className="flex-row justify-between"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        Icon
        <Text
          className="mt-2 text-center font-medium text-white"
        >
          exportAllJson
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        Icon
        <Text
          className="mt-2 text-center font-medium text-white"
        >
          exportAllCsv
        </Text>
      </View>
    </View>
  </View>
  <View
    className="mb-8"
    entering={{}}
  >
    <Text
      className="mb-4 text-lg font-medium text-corp-grey"
    >
      importData
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      Icon
      <Text
        className="mt-2 text-center font-medium text-corp-purple"
      >
        importFromFile
      </Text>
      <Text
        className="mt-1 text-center text-xs text-corp-purple/80"
      >
        supportedFormats
      </Text>
    </View>
  </View>
  <View
    className="mb-8"
    entering={{}}
  >
    <Text
      className="mb-4 text-lg font-medium text-corp-grey"
    >
      onboarding
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      Icon
      <Text
        className="mt-2 text-center font-medium text-corp-teal"
      >
        viewOnboarding
      </Text>
    </View>
  </View>
  <View
    className="mb-8"
    entering={{}}
  >
    <Text
      className="mb-4 text-lg font-medium text-corp-grey"
    >
      dangerZone
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      Icon
      <Text
        className="ml-2 font-semibold text-corp-white"
        style={
          [
            undefined,
          ]
        }
      >
        deleteAllQRCodes
      </Text>
    </View>
  </View>
</View>
`;
